src/ast/effects.h

```diff
@@ -164,10 +169,24 @@ struct EffectAnalyzer : public PostWalker<EffectAnalyzer> {
   }
   void visitLoad(Load *curr) {
     readsMemory = true;
+    isAtomic = curr->isAtomic;
```

Comment:
shouldn't this be `|=`? if it was already marked as atomic, we shouldn't clear the flag